### PR TITLE
Fix FullStatus extra queries on spaces on link layer devices

### PR DESCRIPTION
**Pull request by howbazaar**
During status the code that gets the space name for the link layer device was causing extra queries to the database.

In order to test, John and I introduce a query tracker on the state object to cound the number of database read queries.

### CODE REVIEW

File: apiserver/facades/client/client/status_test.go [hunk 1]
```diff
@@ -101,6 +101,86 @@ func (s *statusSuite) TestFullStatusUnitLeadership(c *gc.C) {
...
+	c.Assert(err, jc.ErrorIsNil)
+
+	s.WaitForModelWatchersIdle(c, s.State.ModelUUID())
+	tracker.Reset()
+
+	_, err = client.Status(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// The number of queries should be the same.
+	c.Check(tracker.ReadCount(), gc.Equals, queryCount)
```
Comment (jameinel) [inline apiserver/facades/client/client/status_test.go]: Would it be worth using gc.Commentf to give context about what queries were done in the case that queryCount doesn't match? Ideally you would have a delta of the previous queries vs the current queries. I guess you can do that when you actually need to debug the failure.

File: state/querytracker.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import "runtime/debug"
+
+// QueryTracker provides a way for tests to determine how many
+// database queries have been made, and who made them.
+type QueryTracker interface {
+	Reset()
+	ListQueries() []QueryDetails
+	ReadCount() int
+	WriteCount() int
```
Comment (jameinel) [inline state/querytracker.go]: we don't currently make use of WriteCount, is it worth postponing it?

File: state/querytracker.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			count++
+		}
+	}
+	return count
+}
+
+// TrackRead records the read query against the collection specified
+// and where the call came from.
+func (q *queryTracker) TrackRead(collectionName string, query interface{}) {
+	q.queries = append(q.queries, QueryDetails{
```
Comment (jameinel) [inline state/querytracker.go]: We are missing direct unit tests of QueryTracker itself, but given it is testing infrastructure, I could be convinced of that.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**